File: Libs_FrameWork/src/main/java/com/ssf/framework/main/adapter/BaseViewHolder.java
package com.ssf.framework.main.adapter;

import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;
import android.view.View;

import com.ssf.framework.autolayout.utils.AutoUtils;


/**
 * @author 小民
 * @time 2017/7/15
 */
public class BaseViewHolder extends RecyclerView.ViewHolder {
    private SparseArray<View> views  = new SparseArray<>();

    public BaseViewHolder(View itemView) {
        super(itemView);
        AutoUtils.autoSize(itemView);
    }

    public <T extends View> T get(int viewId) {
        View view = views.get(viewId);
        if (view == null) {
            view = itemView.findViewById(viewId);
            views.put(viewId, view);
        }
        return (T) view;
    }

}

File: Libs_RefreshLayout/src/main/java/com/ssf/framework/refreshlayout/adapter/BaseViewHolder.kt
package com.ssf.framework.refreshlayout.adapter

import android.support.v7.widget.RecyclerView
import android.util.SparseArray
import android.view.View
import com.ssf.framework.autolayout.utils.AutoUtils

/**
 * @author yedanmin
 * @data 2018/1/17 15:51
 * @describe
 */
class BaseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val views = SparseArray<View>()

    init {
        AutoUtils.autoSize(itemView)
    }

    operator fun <T : View> get(viewId: Int): T {
        var view: View? = views.get(viewId)
        if (view == null) {
            view = itemView.findViewById(viewId)
            views.put(viewId, view)
        }
        return view as T
    }

}